apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -1,35 +1,93 @@
-import {createSlice, PayloadAction} from '@reduxjs/toolkit';
-import {ChatCompletionMessage} from '../types';
+import {createSlice, PayloadAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {
+  ChatCompletionMessage,
+  AichatLevelProperties,
+  Status,
+  Role,
+} from '../types';
 import {initialChatMessages} from '../constants';
 
 const registerReducers = require('@cdo/apps/redux').registerReducers;
+import {LabState} from '@cdo/apps/lab2/lab2Redux';
+import {getChatCompletionMessage} from '../chatApi';
 
 export interface AichatState {
   // All user and assistant chat messages - includes too personal and inappropriate user messages.
   // Messages will be logged and stored.
   chatMessages: ChatCompletionMessage[];
-  // New user message that is submitted via the user chat message editor.
-  newUserMessage: string;
   // Denotes whether we are waiting for a chat completion response from the backend
   isWaitingForChatResponse: boolean;
   // Denotes whether we should show the warning modal
   showWarningModal: boolean;
 }
 
 const initialState: AichatState = {
-  newUserMessage: '',
   chatMessages: initialChatMessages,
   isWaitingForChatResponse: false,
   showWarningModal: true,
 };
 
+// THUNKS
+export const submitChatMessage = createAsyncThunk(
+  'aichat/submitChatMessage',
+  async (message: string, thunkAPI) => {
+    const state = thunkAPI.getState() as {lab: LabState; aichat: AichatState};
+    const systemPrompt = (state.lab.levelProperties as AichatLevelProperties)
+      ?.systemPrompt;
+    if (systemPrompt === undefined) {
+      throw new Error('systemPrompt is undefined');
+    }
```

Comment:
Just a note for later: we may actually want to handle this in AichatView directly, by checking this value whenever `state.lab.levelProperties` updates; whenever the `levelProperties` change, and there is no system prompt, then we can't run the chat app at all and we should probably throw up an error dialog. But this thunk shouldn't need to handle that as it's called on every chat message submission.

Reply:
This makes sense! I'll add a TODO comment here if work is picked up later on.